packages/api/src/external/commonwell/command/cw-directory/rebuild-cw-directory.ts

```diff
@@ -0,0 +1,148 @@
+  const principalAndDelegatesMap = new Map<string, string[]>();
+
+  try {
+    await createTempCwDirectoryTable(sequelize);
+
+    while (!isDone) {
+      try {
+        const maxPosition = currentPosition + BATCH_SIZE;
+        log(`Loading active CW directory entries, from ${currentPosition} up to ${maxPosition}`);
+        const loadStartedAt = Date.now();
+
+        const response = await cw.listOrganizations({
+          offset: currentPosition,
+          limit: BATCH_SIZE,
+          sort: "organizationId",
+        });
+
+        log(`Loaded ${response.organizations.length} entries in ${Date.now() - loadStartedAt}ms`);
+        if (response.organizations.length < BATCH_SIZE) isDone = true;
+
+        const parsedOrgs: CwDirectoryEntryData[] = [];
+        const alreadyInsertedIds = await getCwDirectoryIds(sequelize);
+
+        for (const org of response.organizations) {
+          try {
+            const parsed = parseCWOrganization(org);
+            if (parsed.active) parsedOrgs.push(parsed); // we don't want to store inactive organizations
+          } catch (error) {
+            parsingErrors.push(error as Error);
+          }
+        }
+
+        parsedOrgsCount += parsedOrgs.length;
+        log(`Successfully parsed ${parsedOrgs.length} entries`);
+
+        const orgsToInsert = parsedOrgs.filter(
+          org => !alreadyInsertedIds.some(id => id === org.id)
+        );
+        orgsToInsert.forEach(org => {
+          if (org.delegateOids && org.delegateOids.length > 0) {
+            principalAndDelegatesMap.set(org.id, org.delegateOids);
+          }
+        });
+
+        log(`Adding ${orgsToInsert.length} entries in the DB...`);
+        const insertStartedAt = Date.now();
+        await insertCwDirectoryEntries(sequelize, orgsToInsert);
+        log(`Inserted ${orgsToInsert.length} entries in ${Date.now() - insertStartedAt}ms`);
+
+        if (!isDone) await sleep(SLEEP_TIME.asMilliseconds());
+        currentPosition = maxPosition;
+      } catch (error) {
+        isDone = true;
+        const batchErrorMsg = `Error rebuilding CW directory`;
+        log(
+          `${batchErrorMsg}: batch at position ${currentPosition}. Cause: ${errorToString(error)}`
+        );
+        if (!failGracefully) {
+          throw error;
+        }
+      }
+    }
+
+    if (parsingErrors.length > 0) {
+      const msg = `Parsing errors while rebuilding the CW directory`;
+      const errors = parsingErrors.map(error => errorToString(error)).join("; ");
+      log(msg, errors);
+      capture.message(msg, {
+        extra: {
+          context,
+          amountParsed: parsedOrgsCount,
+          amountError: parsingErrors.length,
+          errors,
+        },
+      });
+    }
+  } catch (error) {
+    await deleteTempCwDirectoryTable(sequelize);
+    const msg = `Failed to rebuild the directory`;
+    log(`${msg}, Cause: ${errorToString(error)}`);
+    capture.error(msg, {
+      extra: { context, error },
```

Comment:
IIRC, we stopped sending the whole `error` to Sentry b/c it includes api-keys from headers and it's quite large; I think we just send `errorToString`, no?